File: apps/ruby-sdk/lib/firecrawl/models/document.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # A scraped document returned by scrape, crawl, and batch endpoints.
    class Document
      attr_reader :markdown, :html, :raw_html, :json, :summary,
                  :metadata, :links, :images, :screenshot, :audio,
                  :video, :attributes, :actions, :answer, :highlights,
                  :warning, :change_tracking, :branding

      def initialize(data)
        @markdown = data["markdown"]
        @html = data["html"]
        @raw_html = data["rawHtml"]
        @json = data["json"]
        @summary = data["summary"]
        @metadata = data["metadata"]
        @links = data["links"]
        @images = data["images"]
        @screenshot = data["screenshot"]
        @audio = data["audio"]
        @video = data["video"]
        @attributes = data["attributes"]
        @actions = data["actions"]
        @answer = data["answer"]
        @highlights = data["highlights"]
        @warning = data["warning"]
        @change_tracking = data["changeTracking"]
        @branding = data["branding"]
      end

      def to_s
        title = metadata&.dig("title") || "untitled"
        url = metadata&.dig("sourceURL") || "unknown"
        "Document{title=#{title}, url=#{url}}"
      end
    end
  end
end

File: apps/dot-net-sdk/Firecrawl/Models/Document.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Represents a scraped document returned by the Firecrawl API.
/// </summary>
public class Document
{
    [JsonPropertyName("markdown")]
    public string? Markdown { get; set; }

    [JsonPropertyName("html")]
    public string? Html { get; set; }

    [JsonPropertyName("rawHtml")]
    public string? RawHtml { get; set; }

    [JsonPropertyName("json")]
    public object? Json { get; set; }

    [JsonPropertyName("summary")]
    public string? Summary { get; set; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, object>? Metadata { get; set; }

    [JsonPropertyName("links")]
    public List<string>? Links { get; set; }

    [JsonPropertyName("images")]
    public List<string>? Images { get; set; }

    [JsonPropertyName("screenshot")]
    public string? Screenshot { get; set; }

    [JsonPropertyName("audio")]
    public object? Audio { get; set; }

    [JsonPropertyName("actions")]
    public object? Actions { get; set; }

    [JsonPropertyName("warning")]
    public string? Warning { get; set; }

    [JsonPropertyName("changeTracking")]
    public object? ChangeTracking { get; set; }

    [JsonPropertyName("branding")]
    public object? Branding { get; set; }
}
